### PR TITLE
Standardize OpenVR WMR grip press as DeviceInputType.TriggerPress

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview
It was `TriggerPress` for one hand and `ButtonPress` for the other
#5874 assumes `TriggerPress`, so I standardized on that.

This PR also adds code to allow profiles to double check their default interaction mappings to update when axes / buttons / etc change.

### CODE REVIEW

File: Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMapping.cs [hunk 1]
```diff
@@ -147,10 +145,60 @@ internal void SetDefaultInteractionMapping(bool overwrite = false)
             }
         }
 
+        internal bool UpdateInteractionSettingsFromDefault()
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMapping.cs]: This code is now effectively duplicated here, any way to unify it?

File: Assets/MixedRealityToolkit.Providers/OpenVR/WindowsMixedRealityOpenVRMotionController.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public WindowsMixedRealityOpenVRMotionController(TrackingState trackingState, Ha
         {
             new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
             new MixedRealityInteractionMapping(1, "Spatial Grip", AxisType.SixDof, DeviceInputType.SpatialGrip, MixedRealityInputAction.None),
-            new MixedRealityInteractionMapping(2, "Grip Press", AxisType.SingleAxis, DeviceInputType.ButtonPress, ControllerMappingLibrary.AXIS_12),
+            new MixedRealityInteractionMapping(2, "Grip Press", AxisType.SingleAxis, DeviceInputType.TriggerPress, ControllerMappingLibrary.AXIS_12),
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit.Providers/OpenVR/WindowsMixedRealityOpenVRMotionController.cs]: Thought I had caught this, but possibly only in the XRTK.  Good catch @keveleigh

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset [hunk 3]
```diff
@@ -1134,7 +1134,7 @@ MonoBehaviour:
         id: 0
         description: None
         axisConstraint: 0
-      keyCode: 346
+      keyCode: 338
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset]: Why the button change in the asset when the mappings have not changed?

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset [hunk 4]
```diff
@@ -1360,7 +1360,7 @@ MonoBehaviour:
         id: 0
         description: None
         axisConstraint: 0
-      keyCode: 347
+      keyCode: 339
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset]: Why the button change in the asset when the mappings have not changed?

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset (hunk 3 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset]: This was left over when you reverted a change to the mappings during the review of #5265 but didn't update the profiles as well.

File: Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMapping.cs (hunk 1 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMapping.cs]: Refactored out `GetDefaultInteractionMappings()`

File: Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset (hunk 3 above)
Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityControllerMappingProfile.asset]: Just doubled checked by creating a new asset and you are indeed correct, 338/339 are the correct values. Great catch.

**Pull request merged**